src/jdk.security.auth/share/classes/com/sun/security/auth/module/GssLoginModule.java

```diff
@@ -0,0 +1,656 @@
+                                   le.getMessage());
+            }
+            if (useFirstPass)
+                throw le;
+        }
+
+        // The first password didn't work or we didn't try it, try prompting
+        try {
+            attemptAuthentication(false);
+            if (debug)
+                System.out.println("\t\t[GssLoginModule] " +
+                                   "authentication succeeded");
+            succeeded = true;
+            cleanState();
+            return true;
+        } catch (LoginException le2) {
+            cleanState();
+            if (debug) {
+                System.out.println("\t\t[GssLoginModule] " +
+                                   (tryFirstPass ? "tryFirstPass " : "") +
+                                   "authentication failed with:" +
+                                   le2.getMessage());
+            }
+            throw le2;
+        }
+    }
+
+    private void getcreds() throws GSSException {
+        if (initiate) {
+            if (debug)
+                System.out.println("\t\t[GssLoginModule] acquiring" +
+                    ((gssName == null) ? " default" : "") +
+                    " initiator credentials...");
+            gssICred = manager.createCredential(gssName, password,
+                    GSSCredential.DEFAULT_LIFETIME, (Oid[])null,
+                    GSSCredential.INITIATE_ONLY);
+            if (debug)
+                System.out.println("\t\t[GssLoginModule] acquired" +
+                    " initiator credentials: " + gssName);
+        }
+        if (accept) {
+            if (debug)
+                System.out.println("\t\t[GssLoginModule] acquiring" +
+                    ((gssName == null) ? " default" : "") +
+                    " acceptor credentials...");
+            gssACred = manager.createCredential(gssName, password,
+                    GSSCredential.DEFAULT_LIFETIME, (Oid[])null,
+                    GSSCredential.ACCEPT_ONLY);
+            // Default acceptor credentials retain a null name
+            if (debug)
+                System.out.println("\t\t[GssLoginModule] acquired" +
+                    " acceptor credentials");
+        }
+        if (gssName == null && gssICred != null)
+            gssName = gssICred.getName();
+        if (gssName == null && gssACred != null)
+            gssName = gssACred.getName();
+    }
+
+    private void attemptAuthentication(boolean getPasswdFromSharedState)
+        throws LoginException {
+
+        // Get a name, maybe
+        if (name == null) {
+            if (useDefaultCreds) {
+                try {
+                    getcreds();
+                    return;
+                } catch (GSSException e) {
+                    throw new LoginException(e.getMessage());
+                }
+            }
+            if (tryDefaultCreds) {
+                try {
+                    getcreds();
+                    return;
+                } catch (GSSException e) { }
```

Comment:
Ignoring exceptions is a bad smell. Add a comment explaining why it's ok.

Reply:
I guess it's obvious to me why: the name of the option, tryDefaultCreds, means that not having a credential means we'll do something else.  There might be other reasons for an exception here than "no credentials", but we'll run into the same again subsequently, and anyways, if you have "try / fallback" logic and you end up with two exceptions, which should you throw back?

I'll add a comment.